src/seedu/addressbook/ui/Formatter.java

```diff
@@ -0,0 +1,164 @@
+    	String[] message = new String[] { DIVIDER, DIVIDER, MESSAGE_WELCOME, version, MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE,
+    		String.format(MESSAGE_USING_STORAGE_FILE, storageFilePath), DIVIDER };
+    	return arrayToString(message);
+    }
+    
+    /**
+     * Returns a formatted goodbye message.
+     * 
+     * @return	message		formatted message
+     */
+    public static String formatGoodbyeMessage() {
+    	String[] message = new String[] { LINE_PREFIX, MESSAGE_GOODBYE, DIVIDER, DIVIDER };
+    	return arrayToString(message);
+    }
+    
+    /**
+     * Returns a formatted failed message.
+     * 
+     * @return	message		formatted message
+     */
+    public static String formatInitFailedMessage() {
+    	String[] message = new String[] { LINE_PREFIX, MESSAGE_INIT_FAILED, DIVIDER, DIVIDER };
+    	return arrayToString(message);
+    }
+    
+    /**
+     * Returns a formatted command result message.
+     * 
+     * @param	result		a CommandResult object with result details
+     * @return	message		formatted message
+     */
+    public static String formatResultToUser(CommandResult result) {
+    	String[] message = new String[] { result.getFeedbackToUser(), DIVIDER };
+    	return arrayToString(message);
+    }
+    
+    /**
+     * Converts an array of strings to a formatted message with line prefixes and separators.
+     * 
+     * @param	components	an array of message components
+     * @return	message		formatted string containing the components
+     */
+    private static String arrayToString(String[] components) {
+    	Formatter.stringBuilder.setLength(0);
+    	for (String c : components) {
+    		Formatter.stringBuilder.append(Formatter.formatLine(c));
+    	}
+    	return Formatter.stringBuilder.toString();
+    }
+    
+    /**
+     * Returns a formatted line string for TextUi.showToUser.
+     * 
+     * @param	component	a component of a message, an unformatted line  
+     * @return	line		formatted line
+     */
+    public static String formatLine(String component) {	
+        return LINE_PREFIX + component.replace("\n", LS + LINE_PREFIX);
+    }
+    
+    /** Formats a list of strings as a viewable indexed list. */
+    public static String getIndexedListForViewing(List<String> listItems) {
+        final StringBuilder formatted = new StringBuilder();
+        int displayIndex = 0 + DISPLAYED_INDEX_OFFSET;
+        for (String listItem : listItems) {
+            formatted.append(getIndexedListItem(displayIndex, listItem)).append("\n");
+            displayIndex++;
+        }
+        return formatted.toString();
+    }
+    
+    /**
+     * Formats a string as a viewable indexed list item.
+     *
+     * @param	visibleIndex	visible index for this listing
+     */
+    private static String getIndexedListItem(int visibleIndex, String listItem) {
+        return String.format(MESSAGE_INDEXED_LIST_ITEM, visibleIndex, listItem);
+    }
+}
```

Comment:
Good job trying to extract as much formatting-related stuff as possible to this class :+1: